models/base.py

```diff
@@ -430,12 +443,17 @@ def get_name_in_current_ctx(self) -> str:
     cur_scope_abs = cur_scope.get_abs_name_ctx_list()
     self_name_abs = self.get_abs_name_ctx_list()
     assert cur_scope_abs[0] is self_name_abs[0]  # same root
-    assert len(cur_scope_abs) >= len(self_name_abs)  # not implemented otherwise
     common_len = 0
-    while cur_scope_abs[common_len + 1] is self_name_abs[common_len + 1]:
+    max_common_len = min(len(cur_scope_abs), len(self_name_abs))
+    while common_len + 1 < max_common_len and cur_scope_abs[common_len + 1] is self_name_abs[common_len + 1]:
       common_len += 1
-    assert common_len == len(self_name_abs) - 2  # not implemented otherwise
-    return "base:" * (len(cur_scope_abs) - len(self_name_abs) + 1) + self.name
+    full_name = self_name_abs[-1].get_abs_name()
+    subs = full_name.split("/")
+    relative_name = subs[common_len:-1]
+    prefix = "/".join(relative_name)
+    if prefix:
+      prefix += "/"
+    return "base:" * (len(cur_scope_abs) - len(self_name_abs) + 1) + prefix + self.name
 
```

Comment:
I also think `common_len` must be used for the calculation of number of `base:`. Again, see the code of `relpath`.

I think sth like this should work (without testing it):

```suggestion
    prefix = "base:" * (len(cur_scope_abs) - common_len + 1)
    postfix = "/".join([ctx.name for ctx in self_name_abs[common_len:]])
    return prefix + postfix

```